lib/xcodeproj/project/object/target_dependency.rb

```diff
@@ -34,9 +34,10 @@ class PBXTargetDependency < AbstractObject
         #
         attribute :platform_filters, Array
 
-        # @return [String] the product reference for this target dependency.
+        # @return [XCSwiftPackageProductDependency] the Swift Package product
+        #         for this target dependency.
         #
-        attribute :product_ref, String
+        has_one :product_ref, XCSwiftPackageProductDependency
```

Comment:
I had to change `attribute` to `has_one` to easily get the product name in the `display_name` method.